packages/neon_framework/lib/src/utils/adaptive.dart

```diff
@@ -17,3 +20,21 @@ bool isCupertino(BuildContext context) {
       return true;
   }
 }
+
+/// Returns whether the current platform on which the framework is currently
+/// executing is a Cupertino one.
+///
+/// This is true for both `TargetPlatform.iOS` and `TargetPlatform.macOS`.
+/// Use [isCupertino] to check against the platform of the current theme.
+bool isCupertinoPlatform() {
```

Comment:
Do we really need a separate method? Just getting this from the theme should be enough?

Reply:
Not really.
The theme could have `TargetPlatform.iOS` while not running on an actual iOS platform. In that case we wouldn't show any icon in release mode because the font isn't bundled (licensing thingy again).

Also how would the syntax be `AdaptiveIcons(context).search` or `AdaptiveIcons.search(context)` or `NeonIcons.adaptive(context).search`?
I find these proposed apis ugly and we would need to see how tree shaking handles this (the first example would be problematic as it isn't using static attributes).

Reply:
I agree that adding the context is ugly, how about only using this method and removing the other one?

Reply:
I also thought about this (maybe even giving us some performance improvement as the platform will not change at runtime).
But as seen by my debugging leftovers above it's quite easy to do some testing through the theme. Especially in our unit tests where we rely on the ability to `mock the platform`.